src/main/java/org/jabref/gui/DefaultInjector.java

```diff
@@ -27,9 +27,7 @@
      * Dependencies without default constructor are constructed by hand.
      */
     private static Object createDependency(Class<?> clazz) {
-        if (clazz == DialogService.class) {
-            return new FXDialogService();
-        } else if (clazz == TaskExecutor.class) {
+         if (clazz == TaskExecutor.class) {
```

Comment:
Please don't remove the injection. Instead of a new `FXDialogService` you can here return `JabRefGUI.getMainFrame().getDialogService()`.

Reply:
Yep, please don't remove injection. This is very handy and makes it otherwise very complicated

Reply:
Ok, second try. I reverted the removal of the injection. But it was not possible to simply call `JabRefGUI.getMainFrame().getDialogService()` inside DefaultInjector, because at the first time it is needed, `JabRefFrame` is not ready yet. In fact, from what i concluded, the injection is used somwhere inside the `init`-method of `JabRefFrame` which was called inside its constructor.
Therefore, i now splitted the constructor of `JabRefFrame` and the `init`-method, so that in `JabRefGUI` the object ist first created and then `init` is called. I'm not so happy with setting `init` `public`, but found no other way.. I also tried to use a factory method, but the constructed `JabRefFrame`-object is needed in `JabRefGUI`, so that `JabRefGUI.getJabRefFrame().getDialogService()`works.
Finally, i also removed the second DialogService in `JabRefGUI`, so that there is only a single DialogService.

Reply:
And you need the JabRefFrame as container for the JFXSnackbar? So that was the initial reason for the sitting of the notify method in BasePanel.

Reply:
Yes, well the reason for the output method in JabRefFrame. So i have to admit, i broke (some of) the notifications with #4815, but it was also confusing with the different methods and the deprecated method in JabRefFrame. So i hope, the code is a little better now.